lib/scout_apm/db_query_metric_set.rb

```diff
@@ -0,0 +1,25 @@
+module ScoutApm
+  class DbQueryMetricSet
+    attr_reader :metrics
+
+    def initialize
+      @metrics = Hash.new
+    end
+
+    def lookup(other)
+      metrics[other.key] ||= other
+    end
+
+    def combine!(other)
+      lookup(other).combine!(other)
+    end
+
+    # Takes an array of ActiveRecord layers, creates new DbQueryMetricStats and combines them into this Set
```

Comment:
Not sure I like this class knowing so much about how we store layers. Elsewhere we stop caring that the data was in layers after the converter step, but in the db case it goes all the way through the Store, into this class.

Can we create those DbQueryMetricStats a lot earlier, and have this class take those as its input?